lib/modules/manager/gleam/extract.ts

```diff
@@ -53,7 +58,55 @@ function extractGleamTomlDeps(gleamToml: GleamToml): PackageDependency[] {
   );
 }
 
-export function extractPackageFile(content: string): PackageFileContent | null {
-  const deps = extractGleamTomlDeps(GleamToml.parse(content));
-  return deps.length ? { deps } : null;
+export async function extractPackageFile(
+  content: string,
+  packageFile: string,
+): Promise<PackageFileContent | null> {
+  let gleamToml: GleamToml;
+  try {
+    gleamToml = GleamToml.parse(content);
+  } catch (err) {
+    logger.debug({ err }, 'Error parsing package file content');
+    return null;
+  }
+
+  const deps = extractGleamTomlDeps(gleamToml);
+  if (!deps.length) {
+    logger.debug('No dependencies found in package file content');
+    return null;
+  }
+
+  const packageFileContent: PackageFileContent = { deps };
+
+  const lockFileName = getSiblingFileName(packageFile, 'manifest.toml');
+  if (lockFileName) {
+    logger.debug(
+      `Found lockFile ${lockFileName} for packageFile: ${packageFile}`,
+    );
+
+    const versionsByPackage = await extractLockFileVersions(lockFileName);
+    if (!versionsByPackage) {
+      logger.debug(`Could not extract locked versions from ${lockFileName}.`);
+      return packageFileContent;
+    }
+
+    packageFileContent.lockFiles = [lockFileName];
+
+    for (const dep of packageFileContent.deps) {
+      const packageName = dep.depName!;
+      const versions = coerceArray(versionsByPackage.get(packageName));
+      const lockedVersion = versioning.getSatisfyingVersion(
+        versions,
+        dep.currentValue!,
+      );
```

Comment:
I think this can throw, so it should contain a try catch?

Reply:
The Gleam manager is largely based off the Rust Cargo manager, since they share very similar behaviors and formats for lockfiles, although they are not quite the same. In this case, the [Cargo manager does set some precedent](https://github.com/renovatebot/renovate/pull/25983/files/414bf95853196c361edacd963868e4397bbc043e#r1407989471) for allowing this to throw, since we don't expect trouble here if we encounter invalid version numbers. Like Cargo, I have added a test to ensure this condition is visible -  71b2bd0dc57b725792129a864032a3c8b16a8ed3

However, I am open to adding the try/catch if you feel it is needed.

Reply:
But you're using a diffferent versioning (cargo uses cargo and this uses hex). 

Good to know hey doesn't throw, other versionings are throwing errors on invalid versions.
So no need for a try-catch yet.